bool collision
float64 time_elapsed
bool new_run
